packages/earn-controller/src/EarnController.ts

```diff
@@ -323,43 +312,34 @@ export class EarnController extends BaseController<
     // temporary array of supported chains
     // TODO: remove this once we export a supported chains list from the sdk
     // from sdk or api to get lending and pooled staking chains
-    this.#supportedPooledStakingChains = [
-      CHAIN_IDS.MAINNET,
-      HOODI_TESTNET_CHAIN_ID_HEX,
-    ];
+    this.#supportedPooledStakingChains = [ChainId.ETHEREUM, ChainId.HOODI];
 
     this.#addTransactionFn = addTransactionFn;
 
-    this.#initializeSDK().catch(console.error);
+    this.#selectedNetworkClientId = selectedNetworkClientId;
+
+    this.#initializeSDK(selectedNetworkClientId).catch(console.error);
     this.refreshPooledStakingData().catch(console.error);
     this.refreshLendingData().catch(console.error);
 
-    const { selectedNetworkClientId } = this.messagingSystem.call(
-      'NetworkController:getState',
-    );
-    this.#selectedNetworkClientId = selectedNetworkClientId;
-
     // Listen for network changes
     this.messagingSystem.subscribe(
-      'NetworkController:stateChange',
+      'NetworkController:networkDidChange',
       (networkControllerState) => {
-        if (
-          networkControllerState.selectedNetworkClientId !==
-          this.#selectedNetworkClientId
-        ) {
-          this.#initializeSDK(
-            networkControllerState.selectedNetworkClientId,
-          ).catch(console.error);
-
-          this.refreshPooledStakingVaultMetadata().catch(console.error);
-          this.refreshPooledStakingVaultDailyApys().catch(console.error);
-          this.refreshPooledStakingVaultApyAverages().catch(console.error);
-          this.refreshPooledStakes().catch(console.error);
-
-          // refresh lending data for all chains
-          this.refreshLendingMarkets().catch(console.error);
-          this.refreshLendingPositions().catch(console.error);
-        }
+        this.#initializeSDK(
+          networkControllerState.selectedNetworkClientId,
+        ).catch(console.error);
+
+        // refresh pooled staking data
+        this.refreshPooledStakingVaultMetadata().catch(console.error);
+        this.refreshPooledStakingVaultDailyApys().catch(console.error);
+        this.refreshPooledStakingVaultApyAverages().catch(console.error);
+        this.refreshPooledStakes().catch(console.error);
```

Comment:
I've left these calls the same here but I don't know if it brings much value to refresh PS data on each network change given PS isn't multi-chain? For the Lending calls below it seems to make more sense.

Reply:
@ameliejyc the idea behind this was moreso to keep the data fresh as the PS data can be is visible in the popular networks filtered view while users are switching to other networks unrelated to PS